components/MultiPanel/index.ts

```diff
@@ -1,12 +1,183 @@
 import './styles.css';
 
+/**
+ * A multi-panel view that the user can add any number of 'panels'. 
+ * 'a panel' consists of two elements. even index element begomes heading, 
+ * and odd index element becomes the expandable content.
+ */
 export default class MultiPanel extends HTMLElement {
+
   constructor () {
     super();
+
+    // Watch for children changes.
+    new MutationObserver(() => this._childrenChange())
+      .observe(this, { childList: true });
   }
 
   connectedCallback () {
+    this._childrenChange();
+
+    const children : Element[] = Array.from(this.children);
+    for (let i : number = 0; i < children.length; i++){
```

Comment:
Nit: use `forEach` instead of a hand-written loop. `forEach` is highly optimized for proper arrays in V8, so it’s likely faster, and it still gives you access to the `index` variable (unlike `for`-`of`).

Reply:
(https://v8project.blogspot.com/2017/09/elements-kinds-in-v8.html if you’re interested in the background)

Reply:
Hah, I recommended for-of below. Which is better? Fwiw I think we should be optimising for readability over what Chrome does fastest on a particular day.

Reply:
> Fwiw I think we should be optimising for readability over what Chrome does fastest on a particular day.

To be clear, I fully agree with this, and this matches our general V8 messaging.

In this case, a handwritten `for` loop expresses _implementation_, whereas `for-of` + `entries` or `forEach` would express _intent_. Expressing intent often leads to more readable code, IMHO.

`forEach` likely being faster than `for` in this case is an indirect result of that (and a nice little bonus). By expressing intent rather than implementation, you generally give the JS engine more information to work with, often leading to greater behind-the-scenes optimization potential.

Reply:
While we're on the subject: generally when iterating over children, it's fastest to use traversal:

```js
let child = this.firstElementChild;
while (child) {
  // stuff
  child = child.nextElementSibling;
}
```

I find this offers a nice balance of readability and performance, since it avoids the array casting entirely.